.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
-# Contributing
+# Contribution Guidelines
 
 ![](https://raw.githubusercontent.com/checkstyle/resources/master/img/checkstyle-logos/checkstyle-logo-260x99.png)
 
-* [Reporting issues](https://checkstyle.org/report_issue.html)
-* [How to report a bug?](https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F)
-* [Issue Template](https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE.md)
+---------------------------------
 
-Please provide issue report in the format that we request, EACH DETAIL IS A HUGE HELP.
+**This guide serves to set clear expectations for everyone involved with the project so that
+we can improve it together while also creating a welcoming space for everyone to participate.
+Following these guidelines will help ensure a positive experience for contributors and
+maintainers alike.**
 
-Issues that are not following the guidelines,
-will be processed as last priority or never or simply closed as invalid.
+:octocat: *Thanks for your interest in contributing to CheckStyle!* :octocat:
 
-## Contributing Guide
+### Contents
+- [Getting Started](#rocket-getting-started)
+- [Code of Conduct](#book-code-of-conduct)
+- [Asking Questions](#bulb-asking-questions)
+- [Opening an Issue](#inbox_tray-opening-an-issue)
+- [Reporting Security Issues](#lock-reporting-security-issues)
+- [Submitting Pull Requests](#technologist-submitting-pull-requests)
+- [Code Review](#white_check_mark-code-review)
+- [Google Summer of Code (GSoC)]()
+- [Credits](#coin-credits)
 
-Please note that PRs are looked only for approved issues. Follow the PR guide for opening a PR
+## :rocket: Getting Started
 
-* [Pull Request Template](https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
+ - Please see the [CheckStyle Documentation](https://checkstyle.org/beginning_development.html)
+    for information on how to get started with the project. This includes setting up your
+    development environment, building the project, and running tests.
+ - Take a look at the [Contribution Guidelines](https://checkstyle.org/contributing.html) for
+    on how to contribute to the project.
+ - Select an issue to work on from the
```

Comment:
I think we should have a document for the labels. You mention first/second issue, but don't mention `easy` and `approved`. Both are very important to mention as new users gloss over approved on issues and jump the gun.

If we are getting rid of any old labels like `easy`, then we should do that.

Reply:
`easy` needs to go in my opinion. I have never liked this label, I remember when I started out here and struggled for a long time on an `easy` issue. `easy` is highly subjective to both experience with software development in general AND experience with the project. 

first, second, third, etc. is good. Even something like `good beginner issue` or even `simple` might be ok. `easy` is likely to make people feel discouraged if they are struggling with it (I know that I did).

Reply:
Issue to improve:
https://github.com/checkstyle/checkstyle/issues/14541

Reply:
@nrmancuso Were you thinking adding text for `approved` will be in other issue? I was not seeing it added yet. I can understand moving the other labels to the issue as there is more to work on, but I think `approved` is pretty much solidified and understood from maintainers and it definitely needs to be conveyed to contributors.

Reply:
yeah, good point, done

Reply:
I think the new `Select an issue to work on from the Issues Page` is too bloated for a bullet. Bullets should be shorter and more to the point, maybe 1-2 sentences, 3 at most. I think this clearly shows that this bullet needs to be its own section "Selecting an Issue to Contribute To" (or some form). I think the mention of the approved label should be in the first 2 sentences.